src/test/java/de/mediathekview/mserver/crawler/orf/tasks/OrfFilmDetailTaskTestWithEpisodes.java

```diff
@@ -0,0 +1,215 @@
+        "http://localhost:8589/apasfpd.apa.at/cms-worldwide/online/394d058ac5456e6a83bb1f8dec9296e4/1517170556/2018-01-26_1705_tl_02_AKTUELL-IN-OEST_24-Stunden-Burg__13962830__o__2332651625__s14226905_5__WEB03HD_17263821P_17285221P_Q6A.mp4",
+        "http://localhost:8589/apasfpd.apa.at/cms-worldwide/online/6b186ef64c267758c08e03f4c409f497/1517170556/2018-01-26_1705_tl_02_AKTUELL-IN-OEST_24-Stunden-Burg__13962830__o__2332651625__s14226905_5__WEB03HD_17263821P_17285221P_Q8C.mp4"
+      },
+      {
+        "Verabschiedung",
+        Duration.of(26, ChronoUnit.SECONDS),
+        "",
+        "http://api-tvthek.orf.at/uploads/media/subtitles/0021/25/78463f837357241ee915d4a12ca41f023304f0e1.ttml",
+        "http://localhost:8589/apasfpd.apa.at/cms-worldwide/online/eb8e5b3de7982efb34aaba16c2f9eb51/1517170556/2018-01-26_1705_tl_02_AKTUELL-IN-OEST_Verabschiedung__13962830__o__1244781515__s14226927_7__WEB03HD_17285221P_17291910P_Q4A.mp4",
+        "http://localhost:8589/apasfpd.apa.at/cms-worldwide/online/0c1abe74d42a186915be9ae331f5ebce/1517170556/2018-01-26_1705_tl_02_AKTUELL-IN-OEST_Verabschiedung__13962830__o__1244781515__s14226927_7__WEB03HD_17285221P_17291910P_Q6A.mp4",
+        "http://localhost:8589/apasfpd.apa.at/cms-worldwide/online/960026cc81d733cf6d1dca429e304ce6/1517170556/2018-01-26_1705_tl_02_AKTUELL-IN-OEST_Verabschiedung__13962830__o__1244781515__s14226927_7__WEB03HD_17285221P_17291910P_Q8C.mp4"
+      }
+    };
+  }
+  
+  @Test
+  public void test() throws IOException {
+    setupHeadRequestForFileSize();
+    JsoupMock.mock(REQUEST_URL,
+        "/orf/orf_film_with_several_parts.html");
+
+    Object[][] films = data();
+    
+    final Set<Film> actual = executeTask(EXPECTED_THEME, REQUEST_URL);
+    
+    assertThat(actual, notNullValue());
+    assertThat(actual.size(), equalTo(films.length));
+    
+    actual.forEach(actualFilm -> {
+      Object[] expectedData = getExpectedValues(films, actualFilm.getTitel());
+      assertThat(expectedData, notNullValue());
+      
+      AssertFilm.AssertEquals(actualFilm,
+        Sender.ORF,
+        EXPECTED_THEME,
+        expectedData[INDEX_TITLE].toString(),
+        EXPECTED_TIME,
+        (Duration)expectedData[INDEX_DURATION],
+        expectedData[INDEX_DESCRIPTION].toString(),
+        REQUEST_URL,
+        EXPECTED_GEO,
+        expectedData[INDEX_URL_SMALL].toString(),
+        expectedData[INDEX_URL_NORMAL].toString(),
+        expectedData[INDEX_URL_HD].toString(),
+        expectedData[INDEX_SUBTITLE].toString()
+      );
+    });
+  }
+
+  private Object[] getExpectedValues(final Object[][] aExpectedFilms, final String aActualTitle) {
+    for (Object[] expected : aExpectedFilms) {
+      if (expected[INDEX_TITLE].toString().compareToIgnoreCase(aActualTitle) == 0) {
+        return expected;
+      }
+    }
+    
+    return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty array instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)